gcc/rust/resolve/rust-name-resolver.h

```diff
@@ -438,14 +455,15 @@ class Resolver
       cb (it->second);
   }
 
-  void iterate_type_ribs (std::function<void (Rib *)> cb)
+  void iterate_type_ribs (std::function<bool (Rib *)> cb)
   {
     for (auto it = type_ribs.begin (); it != type_ribs.end (); it++)
       {
 	if (it->first == global_type_node_id)
 	  continue;
 
-	cb (it->second);
+	if (!cb (it->second))
```

Comment:
This is a good refactor change but I think it should be in a separate PR

Reply:
done and merged in #625